test/transform/ReferenceCollection.test.js

```diff
@@ -0,0 +1,266 @@
+    })
+    it('if neither element or it ancestor(s) have class "reference" returns null', () => {
+      document = domino.createDocument(`
+        <sup id="cite_ref-b"><a id='a2' href="#cite_note-b">[6]</a></sup>
+      `)
+      const anchor = document.querySelector('#a2')
+      assert.equal(closestReferenceClassElement(anchor), null)
+    })
+  })
+  describe('.collectRefText()', () => {
+    const collectRefText = ReferenceCollection.test.collectRefText
+    it('extracts expected reference text', () => {
+      const referenceText = collectRefText(document, document.querySelector('#cite_ref-b'))
+      assert.equal(referenceText, '3 4 5')
+    })
+    it('removes `sup[id^=cite_ref], .mw-cite-backlink` elements from reference text', () => {
+      const referenceText = collectRefText(document, document.querySelector('#cite_ref-d'))
+
+      document = domino.createDocument(referenceText)
+
+      const removalSelector = 'sup[id^=cite_ref], .mw-cite-backlink'
+      const matches = document.querySelectorAll(removalSelector)
+
+      assert.equal(matches.length, 0)
+    })
+  })
+  describe('.getFirstChildAnchor()', () => {
+    const getFirstChildAnchor = ReferenceCollection.test.getFirstChildAnchor
+    it('gets first child anchor', () => {
+      document = domino.createDocument(`
+        <div id="container">
+          <a id='a1' href="#a1">1</a>
+          <a id='a2' href="#a2">2</a>
+        </div>
+      `)
+      const container = document.querySelector('#container')
+      assert.equal(getFirstChildAnchor(container).id, 'a1')
+    })
+    it('returns null if no child anchor', () => {
+      document = domino.createDocument('<div id="container"></div>')
+      const container = document.querySelector('#container')
+      assert.equal(getFirstChildAnchor(container), null)
+    })
+  })
+  describe('.getRefTextContainer()', () => {
+    const getRefTextContainer = ReferenceCollection.test.getRefTextContainer
+    it('gets correct reference text container for tapped anchor container', () => {
+      const tappedAnchorContainer = document.querySelector('#cite_ref-d')
+      assert.equal(getRefTextContainer(document, tappedAnchorContainer).id, 'cite_note-d')
+    })
+  })
+  describe('.hasCitationLink()', () => {
+    const hasCitationLink = ReferenceCollection.test.hasCitationLink
+    beforeEach(() => {
+      document = domino.createDocument(`
+        <sup id="cite_ref-a" class="reference"><a id='a1' href="#cite_note-a">[4]</a></sup>
+        <sup id="cite_ref-b" class="reference"><a id='a2' href="#BLA">[5]</a></sup>
+      `)
+    })
+    it('correctly affirms child anchor has citation link', () => {
+      const element = document.querySelector('#cite_ref-a')
+      assert.equal(hasCitationLink(element), true)
+    })
+    it('correctly affirms child anchor does not have citation link', () => {
+      const element = document.querySelector('#cite_ref-b')
+      assert.equal(hasCitationLink(element), false)
```

Comment:
I replaced all these calls to assert.equal with assert.strictEqual since for the same reason we favor `===` to `==`. :+1:

Reply:
Coolio! Thanks!